catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AccessControlAccessPlugin.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.security;
+
+import static org.codice.ddf.catalog.ui.security.AccessControlUtil.accessAdminHasChanged;
+import static org.codice.ddf.catalog.ui.security.AccessControlUtil.accessGroupsHasChanged;
+import static org.codice.ddf.catalog.ui.security.AccessControlUtil.accessIndividualsHasChanged;
+import static org.codice.ddf.catalog.ui.security.AccessControlUtil.attributeToSet;
+import static org.codice.ddf.catalog.ui.security.AccessControlUtil.isAnyObjectNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.types.Security;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.AccessPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.security.SubjectIdentity;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import org.apache.shiro.SecurityUtils;
+import org.apache.shiro.subject.Subject;
+
+public class AccessControlAccessPlugin implements AccessPlugin {
+
+  private final SubjectIdentity subjectIdentity;
+
+  public AccessControlAccessPlugin(SubjectIdentity subjectIdentity) {
+    this.subjectIdentity = subjectIdentity;
+  }
+
+  protected Subject getSubject() {
+    return SecurityUtils.getSubject();
+  }
+
+  private String getSubjectIdentifier() {
+    return subjectIdentity.getUniqueIdentifier(getSubject());
+  }
+
+  // Equivalent to doing a set intersection of the subject with the access-admin list
+  private final Predicate<Metacard> subjectIsAccessAdmin =
+      (newMetacard) ->
+          attributeToSet
+              .apply(newMetacard, Security.ACCESS_ADMINISTRATORS)
+              .contains(getSubjectIdentifier());
+
+  @VisibleForTesting
+  public boolean isAccessControlUpdated(Metacard prev, Metacard updated) {
+    return !isAnyObjectNull(prev, updated)
```

Comment:
This shouldbe null check ANDed with the grouping of 3 logical OR checks

Reply:
Fixed https://github.com/codice/ddf/pull/3655/commits/6841a7c55d0e61e2f636f2a198f96eac58d37efe